fastocr/service.py

```diff
@@ -53,9 +56,62 @@ async def basic_general(self, image: bytes, lang=''):
     async def close(self):
         await self.session.close()
 
+class YoudaoOcr(BaseOcr):
+    API_BASE = 'https://openapi.youdao.com/ocrapi'
+    CURTIME = str(int(time))
+    SALT = str(uuid1)
+
+    def __init__(self, setting: Setting):
+        self._sign = ''
+        self.appid = setting.get('YoudaoOCR', 'app_id') # appKey in Youdao docs
+        self.seckey = setting.get('YoudaoOCR', 'secret_key') # appSecret in Youdao docs
+        self.session = ClientSession()
+
+    @property
+    def sign(self):
+        if self._sign == '':
+            sign, _ = self.get_sign()
+            self._sign = sign
+        return self._sign
+
+    def truncate(self, image: bytes):
+        q = b64encode(image).decode()
+        q_size = len(q)
+        if q is None:
+            return None
+        else:
+            return q if q_size <= 20 else q[0:10] + str(q_size) + q[q_size - 10:q_size]
+
+    def get_sign(self, image: bytes):
+        sign_str = f'{self.app_id}{self.truncate(image)}{self.SALT}{self.CURTIME}{self.seckey}'
+        sign_hash = sha256().update(sign_str)
+        return sha256().hexdigest
+
+    async def basic_general(self, image: bytes, lang=''):
+        data = {
+            'img': b64encode(image).decode(),
+            'langType': lang if lang != '' else 'auto',
+            'detectType': '10012',
+            'imageType': '1',
+            'appKey': self.appid,
+            'salt': self.SALT,
+            'sign': self.sign,
+            'docType': 'json',
+            'signType': 'v3',
+            'curtime': self.CURTIME
```

Comment:
this class will always be singleton when this application running
`CURTIME` needs to update first